src/Routes.js

```diff
@@ -22,30 +20,19 @@ export const paths = {
     sourceManageApps: '/manage_apps/:id'
 };
 
-const Routes = (props) => {
-    const path = props.location.pathname;
-
-    return (
-        <div className='pf-c-page__main pf-l-page__main'>
-            <Suspense fallback={<Loader/>}>
-                <Switch>
-                    <Route exact path={paths.sources} component={SourcesPage} />
-                    <Route exact path={paths.sourcesNew} component={SourcesPage} />
-                    <Route path={paths.sourcesEdit} component={SourcesPage} />
-                    <Route path={paths.sourcesRemove} component={SourcesPage} />
-                    <Route path={paths.sourceManageApps} component={SourcesPage} />
-                    <Route render={() => some(paths, p => p === path) ? null : (<Redirect to={paths.providers} />)} />
-                </Switch>
-            </Suspense>
-        </div>
-    );
-};
-
-Routes.propTypes = {
-    childProps: PropTypes.any.isRequired,
-    location: PropTypes.shape({
-        pathname: PropTypes.string.isRequired
-    }).isRequired
-};
+const Routes = () =>  (
+    <div className='pf-c-page__main pf-l-page__main'>
+        <Suspense fallback={<Loader/>}>
+            <Switch>
+                <Route exact path={paths.sources} component={SourcesPage} />
+                <Route exact path={paths.sourcesNew} component={SourcesPage} />
+                <Route path={paths.sourcesEdit} component={SourcesPage} />
+                <Route path={paths.sourcesRemove} component={SourcesPage} />
+                <Route path={paths.sourceManageApps} component={SourcesPage} />
+                <Route render={() => <Redirect to={paths.sources} />} />
```

Comment:
I think there can be only one route with SourcesPage. Maybe not event the switch just the one SourcesPage.